---
title: "Gene Expression I"
output: html_notebook
---

```{r}
library(ggplot2)
library(data.table)
library(dplyr)
library(tidyr)

gene_counts <- fread("https://raw.githubusercontent.com/ucdavis-bioinformatics-training/2018-June-RNA-Seq-Workshop/master/thursday/all_counts.txt") %>% as.data.frame()
rownames(gene_counts) <- gene_counts$V1
gene_counts <- gene_counts[,-1]
```

```{r}
gene_counts <- gene_counts %>% rownames_to_column("gene") %>% pivot_longer(
  cols = colnames(gene_counts),
  names_to = "sample",
  values_to = "counts"
) %>% filter(counts > 1000)
head(gene_counts)
```

```{r, fig.height=6, fig.width=10}
p <- ggplot(gene_counts, aes(x = sample, y = counts)) + 
  geom_jitter(aes(color = sample), size = 0.5)
```

```{r, fig.height=6, fig.width=10}
p + theme(legend.position = "none") # remove legend
```

```{r, fig.height=6, fig.width=10}
p + theme(legend.position = "none",
          axis.text.x = element_text(angle = 90, vjust = 0.5))
```

```{r, fig.height=6, fig.width=10}
p + theme(legend.position = "none",
          axis.text.x = element_text(angle = 90, vjust = 0.5),
          axis.title.x = element_blank())+
  ylab("Gene Counts")
```

```{r, fig.height=6, fig.width=10}
p + 
  theme_bw() +
  theme(legend.position = "none",
          axis.text.x = element_text(angle = 90, vjust = 0.5),
          axis.title.x = element_blank())+
  ylab("Gene Counts")
```

```{r, fig.height=6, fig.width=10}
p + 
  theme_minimal() +
  theme(legend.position = "none",
          axis.text.x = element_text(angle = 90, vjust = 0.5),
          axis.title.x = element_blank())+
  ylab("Gene Counts")
```

```{r, fig.height=6, fig.width=10}
p + 
  theme_classic() +
  theme(legend.position = "none",
          axis.text.x = element_text(angle = 90, vjust = 0.5),
          axis.title.x = element_blank())+
  ylab("Gene Counts")
```